tests/bin/test_sh_boot.py

```diff
@@ -43,28 +62,16 @@ def expected(*names):
     all_names = OrderedSet(name.render(version_components=2) for name in names)
     if not names:
         all_names.add(current_interpreter_identity.binary_name(version_components=2))
-    all_names.update(
-        [
-            "python3.12",
-            "python3.11",
-            "python3.10",
-            "python3.9",
-            "python3.8",
-            "python3.7",
-            "python3.6",
-            "python3.5",
-            "python2.7",
-            "pypy3.12",
-            "pypy3.11",
-            "pypy3.10",
-            "pypy3.9",
-            "pypy3.8",
-            "pypy3.7",
-            "pypy3.6",
-            "pypy3.5",
-            "pypy2.7",
-        ]
+
+    supported_versions = sorted(
```

Comment:
The switch to the dynamic list is necessitated by the new support for mutating `Requires-Python` metadata for the Pex dist under CI for Python 3.13. In that case the list needs to include python3.13 and pypy3.13 whereas in the nominal production cases it does not.